### PR TITLE
[Hotfix][SQL Transform] SQL transform plugin bug

### PR DESCRIPTION
**Pull request by qianxkun**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by zhilinli123** [comment]
add set Table identifier, used for the source connector that @qianxkun 
`seaTunnelRow.setTableId(inputRow.getTableId());`
PTAL: @liugddx @EricJoy2048 @rewerma

**Comment by zhilinli123** [comment]
Issues:https://github.com/apache/seatunnel/issues/4784

**Comment by liugddx** [comment]
> add set Table identifier, used for the source connector that @qianxkun `seaTunnelRow.setTableId(inputRow.getTableId());` PTAL: @liugddx @EricJoy2048 @rewerma

Can you add a test case?

**Comment by TyrantLucifer** [comment]
@rewerma PTAL.

**Comment by rewerma** [comment]
It can't simply use source table id for SQL transformer like this: `seaTunnelRow.setTableId(inputRow.getTableId());`, It must check where the source tableId used in the all SQL columns . I will fix it soon.

**Comment by rewerma** [comment]
> seaTunnelRow.setTableId(inputRow.getTableId());

Yes，add `seaTunnelRow.setTableId(inputRow.getTableId());`  is OK.

**Comment by zhilinli123** [comment]
> > seaTunnelRow.setTableId(inputRow.getTableId());
> 
> Yes，add `seaTunnelRow.setTableId(inputRow.getTableId());` is OK.

CC @qianxkun

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLEngine.java [hunk 1]
```diff
@@ -182,8 +182,9 @@ public SeaTunnelRow transformBySQL(SeaTunnelRow inputRow) {
 
         // Project
         Object[] outputFields = project(inputFields);
-
-        return new SeaTunnelRow(outputFields);
+        SeaTunnelRow seaTunnelRow = new SeaTunnelRow(outputFields);
+        seaTunnelRow.setRowKind(inputRow.getRowKind());
```
Comment (zhilinli123) [inline seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLEngine.java:204]: Continue to add:
` seaTunnelRow.setTableId(inputRow.getTableId());`

Comment (hailin0) [reply seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/sql/zeta/ZetaSQLEngine.java:204]: good

**Comment by hailin0** [comment]
rebase dev fix ci error

**Comment by zhilinli123** [comment]
Fix:https://github.com/apache/seatunnel/pull/4819

**Comment by ic4y** [comment]
This issue has already been fixed in https://github.com/apache/seatunnel/pull/4819.

**Pull request closed without merging**